src/ext/client-side-templates.js

```diff
@@ -15,18 +15,21 @@ htmx.defineExtension('client-side-templates', {
 
         var handlebarsTemplate = htmx.closest(elt, "[handlebars-template]");
         if (handlebarsTemplate) {
-            var data = JSON.parse(text);
             var templateName = handlebarsTemplate.getAttribute('handlebars-template');
             return Handlebars.partials[templateName](data);
         }
 
         var nunjucksTemplate = htmx.closest(elt, "[nunjucks-template]");
         if (nunjucksTemplate) {
-            var data = JSON.parse(text);
             var templateName = nunjucksTemplate.getAttribute('nunjucks-template');
-            return nunjucks.render(templateName, data);
+            var template = htmx.find('#' + templateName);
```

Comment:
what do you think about having people use nunjucks-template="#foo" for the inline case? so we don't have to try/catch?

Reply:
The reason the try/catch is there is because nunjucks throws if the template doesn't exist with `nunjucks.render()`, so in order to preserve the ability to render with both `.render()` and then fallback to `.renderString()` inline, I could only think of rescuing the initial `.render()` exception and trying `.renderString()` in the catch block. 

If that throws as well, then it doesn't matter because there's no more code to be executed in the block so it has no effect (besides not gracefully handling the error).

Am open to any suggestions here, `try/catch` is not clean but I looked through the nunjucks API for `.getTemplate()` and it seems that function only works if there's been some sort of filesystem environment mounted (but that could be my misunderstanding).